


### Sources of Experimental Data
We leverage two sets of Microarray data, gene expression changes across different brain injury models (fluid percussion vs controlled cortical) in rats [?1], and hippocampal gene/protein expression and cognitive function in rats across the adult lifespan [?2]. Our goal is to quantify the similarity between the molecular mechanisms that underlying brain damage after traumatic injury to the mechanisms that underlie cognitive decline during aging, with an  emphasis on identifying the similarities between early ageing expression changes to accelerated changes in expression due to brain trauma. To this end, we will perform a pathway analyses, also called a gene-set analyses (GSA), to identify sets of genes that are jointly associated. With the goal of gaining a comprehensive representation of the state of the brain cell throughout it response to brain trauma and throughout the aging process. 

There are multiple analytical methods available to us to ascertain this goal, but we currentyl undertemined on the method that we will chose. As observed from network based studies [?4], [?5], [?6] brain gene expression is organized into modules of distinct functional categories, and so we are also considering gene co-expression network analyis similar to Zhang, B., and Horvath, S. [?6]. We are also considering more advanced techniques in deep learning, such as using  artifical neural networks for co-expression analysis, in particular we are interested in the method developed by Narayanan et al. [?7]. Finally, we are considering the classical cluster analyis introdcued by Eisen et al. [?3]. 

Since rat gene sequences have a more complete functional annotation than human gene sequences, we are at an advantage in terms of giving functional meaning to our findings. We aim to expand our analysis with the addition of the expected uncharateized genes, with the goal of making at least provisional attribution of function on a genomic scale. By relating mechanisms of age related cognitive decline and traumatic brain, we may be able to separate detrimental from compensatory effects in gene expression aging changes and link key early events in cognitive dysfunction. Furthermore we may be able to generate unique insights and testable hypotheses on the underlying processes that drive neurodegeneration.


------------------------------------------------
**[?1]** Natale JE, Ahmed F, Cernak I, Stoica B et al. Gene expression profile changes are commonly modulated across models and species after traumatic brain injury. J Neurotrauma 2003 Oct;20(10):907-27. PMID: 14588109 **[?2]** 	Kadish I, Thibault O, Blalock EM, Chen KC et al. Hippocampal and cognitive aging across the lifespan: a bioenergetic shift precedes and increased cholesterol trafficking parallels memory impairment. J Neurosci 2009 Feb 11;29(6):1805-16. PMID: 19211887 **[?3]** Eisen, M.B., Spellman, P.T., Brown, P.O., and Botstein, D. 1998. Cluster analysis and display of genome-wide expression patterns. Proc.Natl. Acad.Sci. 95: 14863–14868.**[?4]** Chen, Y., Zhu, J., Lum, P.Y., Yang, X., Pinto, S., MacNeil, D.J., Zhang, C., Lamb, J., Edwards, S., Sieberts, S.K., et al. (2008). Variations in DNA elucidate molecular networks that cause disease. Nature 452, 429–435. **[?5]** Emilsson, V., Thorleifsson, G., Zhang, B., Leonardson, A.S., Zink, F., Zhu, J., Carlson, S., Helgason, A., Walters, G.B., Gunnarsdottir, S., et al. (2008). Genetics of gene expression and its effect on disease. Nature 452, 423–428. **[?6]** Zhang, B., and Horvath, S. (2005). A general framework for weighted gene co-expression network analysis. Stat. Appl. Genet. Mol. Biol. 4, e17.**[?7]** Narayanan A,  Keedwell EC,  Gamalielsson J, et al. Single-layer artificial neural networks for gene expression analysis, Neurocomputing , 2004, vol. 61 (pg. 217-40)


